dace/codegen/cppunparse.py

```diff
@@ -644,7 +644,7 @@ def _FunctionDef(self, t, infer_type=False):
     def _AsyncFunctionDef(self, t, infer_type=False):
         self._generic_FunctionDef(t, infer_type, is_async=True)
 
-    def _generic_For(self, t, is_async=False):
+    def _generic_For(self, t, is_async=False, infer_type=False):
```

Comment:
See comment for `assert`.

Actually, I recognized that this function has another problem, that is it generates `auto` for loop variables, not good for OpenCL

Reply:
yes, I noticed that too. this is not great for several reasons on other platforms too.